dohclient.go

```diff
@@ -181,7 +186,12 @@ func (d *DoHClient) ResolveGET(q *dns.Msg) (*dns.Msg, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), d.opt.QueryTimeout)
 	defer cancel()
 
-	req, err := http.NewRequestWithContext(ctx, "GET", u, nil)
+	method := http.MethodGet
+	if d.opt.Use0RTT {
```

Comment:
```suggestion
	if d.opt.Use0RTT && opt.Transport == "quic" {
```
Should we add this? To handle a mis-configuration where Use0RTT is used but it's not actually quic, but http2